rust/hyperlane-core/src/traits/ism.rs

```diff
@@ -0,0 +1,15 @@
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use auto_impl::auto_impl;
+
+use crate::{ChainResult, HyperlaneContract};
+
+/// Interface for the Ism chain contract. Allows abstraction over
+/// different chains
+#[async_trait]
+#[auto_impl(&, Box, Arc)]
+pub trait Ism: HyperlaneContract + Send + Sync + Debug {
```

Comment:
To match how we are doing things already, this should be `InterchainSecurityModule`. The file/module should also be renamed. I think it is fine to keep using things like `MultisigIsm` but the main trait should be spelled out by the conventions we have so far.